main.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import time
+
+start_time = time.time()
+
+N = 10
+data = np.loadtxt("data_5.txt", dtype='float')
+
+
+def SME(x1, x2):
```

Comment:
```suggestion
def SME(x1: np.typing.ArrayLike, x2: np.typing.ArrayLike):
```
можно также указывать типы аргументов, чтобы pycharm подсвечивал неправильное использование метода, а также mypy, если его вызывать в консоли или подрубить в vscode, показывал ошибку использования метода

насчёт размерностей уже нужно ассерты писать, да